### PR TITLE
Create view (from input and from file).

### PR DESCRIPTION
**Pull request by pengx17**
- [x] Create from input
- [x] Create from file

Create from settings requires major work time and will be included in a new PR.

### CODE REVIEW

File: src/app/frontend/create/from/form/component.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Component, Input, OnInit} from '@angular/core';
+import {HistoryService} from '../../../common/services/global/history';
+import {NamespaceService} from '../../../common/services/global/namespace';
+import {overviewState} from '../../../overview/state';
+import {CreateService} from '../../service';
+
+@Component(
```
Comment (pengx17) [inline src/app/frontend/create/from/form/component.ts]: This component is work in progress.

File: src/app/frontend/common/services/global/history.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import {Injectable, transition} from '@angular/core';
+import {RawParams, StateService, Transition, TransitionPromise, TransitionService} from '@uirouter/core';
+import {KdStateService} from './state';
+
+@Injectable()
+export class HistoryService {
+  private previousStateName: string;
+  private previousStateParams: RawParams;
+
+  constructor(private readonly state: StateService, private readonly kdState: KdStateService) {
```
Comment (maciaszczykm) [inline src/app/frontend/common/services/global/history.ts]: If some service is private end its name with `_`, ie. `state_`.

File: src/app/frontend/common/services/global/history.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  init(): void {
+    this.kdState.onSuccess.subscribe((transition: Transition) => {
+      this.previousStateName = transition.from().name || '';
+      this.previousStateParams = transition.params('from');
+    });
+  }
+
+  /**
+   * Goes back to previous state or to the provided default if none set.
+   * @param {string} defaultState
```
Comment (maciaszczykm) [inline src/app/frontend/common/services/global/history.ts]: We do not longer use type annotations in comments.

Comment (maciaszczykm) [reply src/app/frontend/common/services/global/history.ts]: Moreover, `back` doesn't explain what the function does.

File: src/app/frontend/create/from/form/component.ts (hunk 1 above)
Comment (maciaszczykm) [reply src/app/frontend/create/from/form/component.ts]: You can add some `TODO` comment.

File: src/app/frontend/create/from/form/component.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    {selector: 'kd-create-from-form', templateUrl: './template.html', styleUrls: ['./style.scss']})
+export class CreateFromFormComponent {
+  readonly namePattern = new RegExp('^[a-z]([-a-z0-9]*[a-z0-9])?$');
+  readonly nameMaxLength = 24;
+
+  // Model
```
Comment (maciaszczykm) [inline src/app/frontend/create/from/form/component.ts]: Remove it.

File: src/app/frontend/create/from/form/helpsection/style.scss [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+@import "../../../../variables";
+
+// This rule is meant to align items only in the current layout (kd-help-section).
```
Comment (maciaszczykm) [inline src/app/frontend/create/from/form/helpsection/style.scss]: Switch to `Align items only in the current layout (kd-help-section).`

File: src/app/frontend/create/from/form/helpsection/userhelp/style.scss [hunk 6]
```diff
@@ -11,10 +11,18 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+@import '../../../../../variables';
```
Comment (maciaszczykm) [inline src/app/frontend/create/from/form/helpsection/userhelp/style.scss]: Missing empty line?

File: src/app/frontend/create/from/form/module.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+@NgModule({
+  declarations: [
+    HelpSectionComponent,
+    UserHelpComponent,
+    CreateFromFormComponent,
+    UniqueNameValidator
+  ],
+  imports: [CommonModule, SharedModule, ComponentsModule],
+  exports: [CreateFromFormComponent],
+  providers: []
```
Comment (maciaszczykm) [inline src/app/frontend/create/from/form/module.ts]: Empty array?

File: src/app/frontend/common/services/global/history.ts (hunk 3 above)
Comment (pengx17) [reply src/app/frontend/common/services/global/history.ts]: will update the changes in #2907. `back` will be changed to 'goBack'

Comment (maciaszczykm) [reply src/app/frontend/common/services/global/history.ts]: `goBack` or `goToPreviousState` seem to be okay for me.

cc @floreks

**Comment by pengx17** [comment]
@maciaszczykm Pr is Updated per review comments.

**Comment by pengx17** [comment]
@floreks one thing I missed is that the create service actually carries global state `isDeployInProgress`. Is it better to be moved into `common/services/global` instead?

**Comment by floreks** [comment]
I don't think `isDeployInProgress` is required globally. Globally means that most of the state components is using a service. If only few components inject a service then is should be kept in a separate module. This way tree shaking and lazy loading modules works better.

**Comment by pengx17** [comment]
@floreks please reset the branch and try compile again. The compilation issue should have been fixed.

File: src/app/frontend/common/services/create/service.ts [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+  /** Confirmation text for the dialog shown on deploy validation error. */
+  MSG_DEPLOY_ANYWAY_DIALOG_OK: 'Yes',
+
+  /** Cancellation text for the dialog shown on deploy validation error. */
+  MSG_DEPLOY_ANYWAY_DIALOG_CANCEL: 'No'
+};
+
+@Injectable()
+export class CreateService {
+  private isDeployInProgress = false;
```
Comment (floreks) [inline src/app/frontend/common/services/create/service.ts]: We are adding suffix `_` to all private members (variables and functions).

**Comment by floreks** [comment]
I have tested it and everything seems to be working fine. Nice work!

**Pull request closed without merging**